tools/flang2/flang2exe/cgmain.cpp

```diff
@@ -13622,31 +13685,26 @@ print_function_signature(int func_sptr, const char *fn_name, LL_ABI_Info *abi,
   print_token(")");
 
   /* Function attributes.  With debugging turned on, the debug attributes
-     contain "noinline", so there is no need to repeat it here. */
+     contain "noinline" (and others), so there is no need to repeat it here. */
   if (need_debug_info(SPTR_NULL)) {
     /* 'attributes #0 = { ... }' to be emitted later */
     print_token(" #0");
-  } else if (!XBIT(183, 0x10) || XBIT(14, 0x8)) {
-    /* Nobody sets -x 183 0x10, besides Flang. We're disabling LLVM inlining for
-     * proprietary compilers. */
-    /* 2nd XBIT - Apply noinline attribute if the pragma "noinline" is given */
-    print_token(" noinline");
-  }
-  if (XBIT(191, 0x2)) {
-    /* Apply alwaysinline attribute if the pragma "forceinline" is given */
-    print_token(" alwaysinline");
-  }
-  if (flg.target_features) {
-    print_token(" \"target-features\"=\"");
-    print_token(flg.target_features);
-    print_token("\"");
-  }
-  if (flg.vscale_range_min && flg.vscale_range_max) {
-    char vsrange[64U];
-    snprintf(vsrange, sizeof vsrange, " vscale_range(%d,%d)",
-                                      flg.vscale_range_min,
-                                      flg.vscale_range_max);
-    print_token(vsrange);
+  } else {
+    if (!XBIT(183, 0x10) || XBIT(14, 0x8)) {
+      /* Nobody sets -x 183 0x10, besides Flang. We're disabling LLVM inlining for
+       * proprietary compilers. */
+      /* 2nd XBIT - Apply noinline attribute if the pragma "noinline" is given */
+      print_token(" noinline");
```

Comment:
It took me some time to realize the problem in [my comment](https://github.com/flang-compiler/flang/pull/1267#discussion_r919984333). The `XBIT(14, 0x8)` and `XBIT(191, 0x2)` tests should be lifted out of this `else` block since they come from pragmas and should only be set on a specific function, not inside the global attribute list `#0`. The other changes seem okay (`fp-contract=fast`, target features, and vscale range are global attributes).

Reply:
modified.